app/src/main/java/com/codesoom/assignment/service/ProductServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package com.codesoom.assignment.service;
+
+import com.codesoom.assignment.domain.Product;
+import com.codesoom.assignment.repository.ProductRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Transactional
+public class ProductServiceImpl implements ProductService {
+
+    private final ProductRepository toyProductRepository;
+
+    public ProductServiceImpl(ProductRepository toyProductRepository) {
+
+        this.toyProductRepository = toyProductRepository;
+
+    }
+
+    @Override
+    public Product register(Product product) {
+
+       return toyProductRepository.save(product);
+
+    }
+
+    @Override
+    public Product getProduct(Long id) {
+
+        return toyProductRepository.findById(id).orElseThrow();
+
+    }
+
+    @Override
+    public List<Product> getProducts() {
+
+        return toyProductRepository.findAll();
+
+    }
+
+    @Override
+    public Product updateProduct(Long id, Product product) {
+
+        Product updateProduct = getProduct(id);
+        updateProduct.setProduct(product);
+
+        return updateProduct;
+
+    }
+
+    @Override
+    public void delete(Long id) {
+
+        Optional<Product> foundProduct = toyProductRepository.findById(id);
+
+        if (foundProduct.isPresent()) {
+
+            toyProductRepository.deleteById(foundProduct.get().getId());
+
+        }
```

Comment:
```suggestion
        toyProductRepository.findById(id)
            .ifPresent(it -> {
                toyProductRepository.deleteById(it.getId());
            })
```

반환값이 없다면 이렇게도 작성할 수 있습니다.

Reply:
멋진 라인인것 같습니다.. 이렇게도 작성할 수있군요 ㅠ 람다식을 적절하게 사용하는 연습을 해보겠습니다